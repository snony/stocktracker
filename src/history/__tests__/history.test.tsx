import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import History, { HistoryChart } from 'history/history'
import React from 'react'
import renderer from 'react-test-renderer'

import {
  mockEmptyHistoryContainerProps,
  mockHistoryChartProps,
  mockHistoryContainerProps
} from './__mock__/mockData'

Enzyme.configure({ adapter: new Adapter() })

const createNodeMock = () => {
  const doc = document.implementation.createHTMLDocument()
  return { parentElement: doc.body }
}

describe('history component', () => {
  it('should render history chart correctly', () => {
    const tree = renderer
      .create(<HistoryChart {...mockHistoryChartProps} />, { createNodeMock })
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly', () => {
    const tree = renderer
      .create(<History {...mockHistoryContainerProps} />, { createNodeMock })
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with no historical data', () => {
    const wrapper = shallow(<History {...mockEmptyHistoryContainerProps} />)

    expect(wrapper.getElement()).toBeNull()
  })

  it('should render filter buttons correctly', () => {
    const wrapper = shallow(<History {...mockHistoryContainerProps} />)

    expect(wrapper.find('FilterButton[type="date"]').length).toBe(6)
    expect(wrapper.find('FilterButton[type="price"]').length).toBe(4)
  })

  describe("should change filter button's selected attribute when filter is changed", () => {
    const wrapper = shallow(<History {...mockHistoryContainerProps} />)

    it('date filter', () => {
      wrapper.setProps({ dateFilter: '1m' })
      expect(wrapper.find('FilterButton[value="1m"]').prop('selected')).toBeTruthy()
    })

    it('price filter', () => {
      wrapper.setProps({ priceFilter: 'close' })
      expect(wrapper.find('FilterButton[value="close"]').prop('selected')).toBeTruthy()
    })
  })

  it('should call onClickFilterHistoryBy[Date|Price] when a filter button is clicked', () => {
    const wrapper = mount(<History {...mockHistoryContainerProps} />)

    wrapper.find('FilterButton[value="1m"]').simulate('click')
    wrapper.find('FilterButton[value="close"]').simulate('click')

    expect(mockHistoryContainerProps.onClickFilterHistoryByDate).toHaveBeenCalledWith('AAPL', '1m')
    expect(mockHistoryContainerProps.onClickFilterHistoryByPrice).toHaveBeenCalledWith(
      'AAPL',
      'close'
    )
  })
})
