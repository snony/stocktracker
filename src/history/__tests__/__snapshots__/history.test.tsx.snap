// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`history component should render correctly 1`] = `
Array [
  .emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

@media screen and (max-width:600px) {
  .emotion-12 {
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

.emotion-4 + .emotion-4 {
  margin-left: 1.5rem;
}

.emotion-0 {
  background-color: transparent;
  border-color: transparent;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  outline: none;
  font-size: 0.9rem;
  margin: 0 0.3rem;
  color: white;
}

.emotion-1 {
  background-color: transparent;
  border-color: transparent;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  outline: none;
  font-size: 0.9rem;
  margin: 0 0.3rem;
}

<div
    className="emotion-12"
  >
    <div
      className="emotion-4"
    >
      <button
        className="emotion-0"
        onClick={[Function]}
        selected={true}
      >
        CLOSE
      </button>
      <button
        className="emotion-1"
        onClick={[Function]}
        selected={false}
      >
        OPEN
      </button>
      <button
        className="emotion-1"
        onClick={[Function]}
        selected={false}
      >
        HIGH
      </button>
      <button
        className="emotion-1"
        onClick={[Function]}
        selected={false}
      >
        LOW
      </button>
    </div>
    <div
      className="emotion-4"
    >
      <button
        className="emotion-0"
        onClick={[Function]}
        selected={true}
      >
        YTD
      </button>
      <button
        className="emotion-1"
        onClick={[Function]}
        selected={false}
      >
        1D
      </button>
      <button
        className="emotion-1"
        onClick={[Function]}
        selected={false}
      >
        1M
      </button>
      <button
        className="emotion-1"
        onClick={[Function]}
        selected={false}
      >
        6M
      </button>
      <button
        className="emotion-1"
        onClick={[Function]}
        selected={false}
      >
        1Y
      </button>
      <button
        className="emotion-1"
        onClick={[Function]}
        selected={false}
      >
        5Y
      </button>
    </div>
  </div>,
  .emotion-0 {
  height: 420px;
  width: 99% !important;
}

<div
    className="emotion-0"
  >
    <div
      className="recharts-responsive-container"
      style={
        Object {
          "height": "100%",
          "maxHeight": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "hidden",
            "zIndex": -1,
          }
        }
      >
        <div
          onScroll={[Function]}
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "zIndex": -1,
            }
          }
        >
          <div
            style={
              Object {
                "height": NaN,
                "left": 0,
                "position": "absolute",
                "top": 0,
                "width": NaN,
              }
            }
          />
        </div>
        <div
          onScroll={[Function]}
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "zIndex": -1,
            }
          }
        >
          <div
            style={
              Object {
                "height": "200%",
                "left": 0,
                "position": "absolute",
                "top": 0,
                "width": "200%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`history component should render correctly before initial fetch 1`] = `
.emotion-0 {
  color: white;
  display: block;
  margin-bottom: 0.3rem;
  font-size: 0.9rem;
}

<span
  className="emotion-0"
>
  Loading...
</span>
`;

exports[`history component should render correctly on data fetch failed 1`] = `
.emotion-0 {
  color: white;
  display: block;
  margin-bottom: 0.3rem;
  font-size: 0.9rem;
}

<span
  className="emotion-0"
>
  Failed to fetch stock history
</span>
`;

exports[`history component should render history chart correctly 1`] = `
<div
  className="recharts-responsive-container"
  style={
    Object {
      "height": "100%",
      "maxHeight": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "overflow": "hidden",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "visibility": "hidden",
        "zIndex": -1,
      }
    }
  >
    <div
      onScroll={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "visibility": "hidden",
          "zIndex": -1,
        }
      }
    >
      <div
        style={
          Object {
            "height": NaN,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": NaN,
          }
        }
      />
    </div>
    <div
      onScroll={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "visibility": "hidden",
          "zIndex": -1,
        }
      }
    >
      <div
        style={
          Object {
            "height": "200%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "200%",
          }
        }
      />
    </div>
  </div>
</div>
`;
